### PR TITLE
mpi bugfix in  test_metis_submodelpart_list.py

### PR DESCRIPTION
**Pull request by pablobecker**
**📝 Description**
This test crashes in MPI if more than 4 cores are used. The unordered dict used to verify the number of entities in the submodelparts was accessed in a different order by the different processors, therefore adding the number of nodes/elems/conds of different model parts.
I simply replaced the dict by a vector, so that the same order is used by all the threads.

The process itself works ok, this just makes the test pass. But it's a good reminder that unordered dicts are , well, unordered, so nothing can be taken for granted

**🆕 Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- Fixes test_metis_submodelpart_list.py in MPI paralellism

### CODE REVIEW

**Comment by pablobecker** [comment]
ping @RiccardoRossi

File: applications/MetisApplication/tests/test_metis_submodelpart_list.py [hunk 1]
```diff
@@ -41,23 +41,25 @@ def test_ReadTwoSubModelParts(self):
             },
             "echo_level" : 0
         }""")
-        results = {"Main.submodelpart_liquid" : [133, 381, 228],
-                   "Main.submodelpart_solid" :  [280, 810, 552]}
+        results = [["Main.submodelpart_liquid" , [133, 381, 228]],
+                   ["Main.submodelpart_solid" ,  [280, 810, 552]]]
         ReadModelPart(self.file_name, model_part, settings)
         for submodel_part_name in results:
+            submodel_part_name = i_result[0]
```
Comment (RiccardoRossi) [inline applications/MetisApplication/tests/test_metis_submodelpart_list.py:48]: i think you also need to change the for...

~~~
for i_result in results:
~~~

File: applications/MetisApplication/tests/test_metis_submodelpart_list.py [hunk 2]
```diff
@@ -86,24 +88,26 @@ def test_ReadSubSubModelParts(self):
...
         }""")
-        results = {"Main.submodelpart_liquid.ingate" : [81, 188, 110],
-                   "Main.submodelpart_liquid.mainPart" : [85, 193, 118],
-                   "Main.submodelpart_solid" : [280,810,552]}
+        results = [["Main.submodelpart_liquid.ingate" , [81, 188, 110]],
+                   ["Main.submodelpart_liquid.mainPart" , [85, 193, 118]],
+                   ["Main.submodelpart_solid" , [280,810,552]]]
         ReadModelPart(self.file_name, model_part, settings)
         for submodel_part_name in results:
+            submodel_part_name = i_result[0]
```
Comment (RiccardoRossi) [inline applications/MetisApplication/tests/test_metis_submodelpart_list.py:96]: same in the line above to this one... you miss changing the loop

File: applications/MetisApplication/tests/test_metis_submodelpart_list.py [hunk 3]
```diff
@@ -86,24 +88,26 @@ def test_ReadSubSubModelParts(self):
...
-            self.assertTrue(local_number_nodes > 0)
-            self.assertTrue(local_number_elements > 0)
-            self.assertTrue(local_number_conditions > 0)
+            if self.size<=10: #if too many partitions are used, some may end up with no nodes/elems
+                self.assertTrue(local_number_nodes > 0)
+                self.assertTrue(local_number_elements > 0)
+                self.assertTrue(local_number_conditions > 0)
             total_nodes = submodel_part.GetCommunicator().GetDataCommunicator().SumAll(local_number_nodes)
             total_elements =submodel_part.GetCommunicator().GetDataCommunicator().SumAll(local_number_elements)
             total_conditions = submodel_part.GetCommunicator().GetDataCommunicator().SumAll(local_number_conditions)
```
Comment (philbucher) [inline applications/MetisApplication/tests/test_metis_submodelpart_list.py]:
```suggestion
            total_nodes = submodel_part.GetCommunicator().GlobalNumberOfNodes()
            total_elements = submodel_part.GetCommunicator().GlobalNumberOfElements()
            total_conditions = submodel_part.GetCommunicator().GlobalNumberOfConditions()

```

**Comment by pablobecker** [comment]
thanks phil, pushed your suggestion

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**